From 89a6eeaa6ceaa592cfbda6287342cd52c533495f Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Tue, 9 Jan 2018 16:31:07 +0100
Subject: [PATCH 32/84] Revert "scripts/package-metadata.pl: inhibit compile
 deps on missing build types"

This reverts commit b616aa6db7a9952c182ab49d7942fb67c09803d5. The change
conflicts with our PROVIDES fix and is not strictly necessary for Gluon.
---
 scripts/package-metadata.pl | 31 ++++++-------------------------
 1 file changed, 6 insertions(+), 25 deletions(-)

diff --git a/scripts/package-metadata.pl b/scripts/package-metadata.pl
index c49d132912..2da32c770e 100755
--- a/scripts/package-metadata.pl
+++ b/scripts/package-metadata.pl
@@ -460,27 +460,20 @@ sub gen_package_mk() {
 			next unless $pkg->{"builddepends/$type"};
 			foreach my $dep (@{$pkg->{"builddepends/$type"}}) {
 				my $suffix = "";
-				my $deptype = "";
 				my $condition;
 
 				if ($dep =~ /^(.+):(.+)/) {
 					$condition = $1;
 					$dep = $2;
 				}
-				if ($dep =~ /^(.+)\/(.+)/) {
+				if ($dep =~ /^(.+)(\/.+)/) {
 					$dep = $1;
-					$deptype = $2;
-					$suffix = "/$2";
+					$suffix = $2;
 				}
 
 				my $idx = "";
 				my $pkg_dep = $package{$dep};
 				if (defined($pkg_dep) && defined($pkg_dep->{src})) {
-					unless (!$deptype || grep { $_ eq $deptype } @{$pkg_dep->{buildtypes}}) {
-						warn sprintf "WARNING: Makefile '%s' has a %s build dependency on '%s/%s' but '%s' does not implement a '%s' build type\n",
-							$pkg->{makefile}, $type, $pkg_dep->{src}, $deptype, $pkg_dep->{makefile}, $deptype;
-						next;
-					}
 					$idx = $pkg_dep->{subdir}.$pkg_dep->{src};
 				} elsif (defined($srcpackage{$dep})) {
 					$idx = $subdir{$dep}.$dep;
@@ -506,16 +499,14 @@ sub gen_package_mk() {
 			my $condition;
 			my $prefix = "";
 			my $suffix = "";
-			my $deptype = "";
 
 			if ($deps =~ /^(.+):(.+)/) {
 				$condition = $1;
 				$deps = $2;
 			}
-			if ($deps =~ /^(.+)\/(.+)/) {
+			if ($deps =~ /^(.+)(\/.+)/) {
 				$deps = $1;
-				$deptype = $2;
-				$suffix = "/$2";
+				$suffix = $2;
 			}
 
 			my $pkg_dep = $package{$deps};
@@ -530,17 +521,7 @@ sub gen_package_mk() {
 			foreach my $dep (@deps) {
 				$pkg_dep = $package{$deps};
 				if (defined $pkg_dep->{src}) {
-					unless (!$deptype || grep { $_ eq $deptype } @{$pkg_dep->{buildtypes}}) {
-						warn sprintf "WARNING: Makefile '%s' has a build dependency on '%s/%s' but '%s' does not implement a '%s' build type\n",
-							$pkg->{makefile}, $pkg_dep->{src}, $deptype, $pkg_dep->{makefile}, $deptype;
-						next;
-					}
-					unless ($pkg->{src} ne $pkg_dep->{sec}.$suffix) {
-						warn sprintf "WARNING: Makefile '%s' has a build dependency on itself\n",
-							$pkg->{makefile};
-						next;
-					}
-					$idx = $pkg_dep->{subdir}.$pkg_dep->{src};
+					($pkg->{src} ne $pkg_dep->{src}.$suffix) and $idx = $pkg_dep->{subdir}.$pkg_dep->{src};
 				} elsif (defined($srcpackage{$dep})) {
 					$idx = $subdir{$dep}.$dep;
 				}
@@ -592,7 +573,7 @@ ifndef DUMP_TARGET_DB
 	( \\
 $cmds \\
 	) > \$@
-
+	
 ifneq (\$(IMAGEOPT)\$(CONFIG_IMAGEOPT),)
   package/preconfig: \$(TARGET_DIR)/etc/uci-defaults/$preconfig
 endif
-- 
2.17.1

